Classes/Emogrifier.php

```diff
@@ -952,6 +939,44 @@ private function removeInvisibleNodes(\DOMXPath $xPath)
     }
 
     /**
+     * Before be begin processing the CSS file, parse the document and normalize all existing CSS attributes.
+     * This changes 'DISPLAY: none' to 'display: none'.
+     * We wouldn't have to do this if DOMXPath supported XPath 2.0.
+     * Also store a reference of nodes with existing inline styles so we don't overwrite them.
+     *
+     * @param \DOMXPath $xPath
+     *
+     * @return void
+     *
+     * @throws \RuntimeException
+     */
+    private function normalizeStyleAttributesOfAllNodes(\DOMXPath $xPath)
+    {
+        try {
+            $nodesWithStyleAttribute = $this->getAllNodesWithStyleAttribute($xPath);
```

Comment:
This is the only part that could cause the exception if I'm reading this correctly. So we could move the exception handling to directly below/around this line, moving the data wrangling out of the error handling.